src/Adapter/MSTestAdapter.PlatformServices/Services/TestSourceHost.cs

```diff
@@ -83,7 +83,7 @@ internal TestSourceHost(string sourceFileName, IRunSettings? runSettings, IFrame
         SetContext(sourceFileName);
 
         // Set isAppDomainCreationDisabled flag
-        _isAppDomainCreationDisabled = _runSettings != null && MSTestAdapterSettings.IsAppDomainCreationDisabled(_runSettings.SettingsXml);
+        _isAppDomainCreationDisabled = _runSettings == null ? true : MSTestAdapterSettings.IsAppDomainCreationDisabled(_runSettings.SettingsXml);
```

Comment:
You can simplify this line:

```suggestion
        _isAppDomainCreationDisabled = MSTestAdapterSettings.IsAppDomainCreationDisabled(_runSettings?.SettingsXml);
```